src/components/App.js

```diff
@@ -27,15 +27,23 @@ const WelcomeBox = createFactory(require("./WelcomeBox").default);
 const EditorTabs = createFactory(require("./Editor/Tabs"));
 
 class App extends Component {
-  state: { horizontal: boolean };
+  state: {
+    horizontal: boolean,
+    startPanelSize: number,
+    endPanelSize: number,
+  };
   onLayoutChange: Function;
   getChildContext: Function;
   renderEditorPane: Function;
   renderVerticalLayout: Function;
 
   constructor(props) {
     super(props);
-    this.state = { horizontal: verticalLayoutBreakpoint.matches };
+    this.state = {
+      horizontal: verticalLayoutBreakpoint.matches,
+      startPanelSize: 0,
+      endPanelSize: 0,
```

Comment:
I think @wldcordeiro might recommend that we put this state in the UI state. 

What do you think?